tools/toolchains/arm.py

```diff
@@ -70,8 +76,12 @@ def __init__(self, target, notify=None, macros=None,
         if target.core not in self.SUPPORTED_CORES:
             raise NotSupportedException(
                 "this compiler does not support the core %s" % target.core)
+            
+        if  getattr(target, "default_lib", "std") == "small" and getattr(target, "microlib_supported", False) == False:
+            raise NotSupportedException(
+                  "MicroLib is not supported for this target")
```

Comment:
It is not Microlib that is not supported. It is the option default_lib that is not supported for the target.
```suggestion
                  "target.default_lib" = "small" is not supported for this target")
```